common/simple_dom.h

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2022 The Photon Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+#include <inttypes.h>
+#include <assert.h>
+#include <memory>
+#include <math.h>
+#include <atomic>
+#include <photon/common/object.h>
+#include <photon/common/estring.h>
+#include <photon/common/enumerable.h>
+#include <photon/common/stream.h>
+
+
+namespace photon {
+
+namespace fs {
+class IFileSystem;
+}
+
+// simple unified interface for common needs in reading
+// structured documents such as xml, json, yaml, ini, etc.
+namespace SimpleDOM {
+
+// SimpleDOM emphasize on:
+// 1. simple & convenient usage;
+// 2. unified interface, common needs (may not fit for corner cases);
+// 3. efficient parsing (reading), as well as compiling;
+
+using str = estring_view;
+
+class RootNode;
+struct NodeWrapper;
+
+// the interface for internal implementations
```

Comment:
Can it be concealed for internal use and not disclosed externally?

Reply:
No，because it's also the binary interface.